### DISCUSSION TITLE
delete tested data

### DISCUSSION THREAD

**Discussion by 0MAL** [mosparo/mosparo#219] [Q&A]
Can I delete submission data by just deleting rows of testing posts from submission table in database?

**Comment by zepich** [comment]
Hi @0MAL 

Thank you very much for your question.

In general, you could delete rows in the `submission` and the `submit_token` tables. But technically, this is not required since the data is gone anyway after two weeks.

Or why do you want to delete the data? Do you have any issue?

Kind regards,

zepich

**Reply by 0MAL** [reply]
Thanks. I just think that test data is confusing and should be deleted in the production environment.

**Reply by zepich** [reply]
Hi @0MAL 

Okay, thank you very much for your feedback.

Right now, there are only three solutions for this:

### Option 1

Delete the rows manually from the database. It should be only in `submission` and `submit_token` tables.

### Option 2

The other option is to adjust and execute the cleanup script once. This method only works if you know that all test data of your mosparo installation was created before a specific date, for example, before 2024-04-12. If that's the case, you could adjust the file `src/Helper/CleanupHelper.php` and change the number `P14D` to (for example) `P3D` (see here: https://github.com/mosparo/mosparo/blob/master/src/Helper/CleanupHelper.php#L92).

When executing the cleanup command via CLI (for example, `php bin/console mosparo:cleanup-database`), it will delete all submissions and submit tokens before these 3 days, so before the 12th of April.

After the cleanup, you must return the value to `P14D`. Otherwise, your submissions will always be deleted after 3 days.

### Option 3

You can also export a project's settings and delete it. Then, you can create a new project and import the exported settings.

Important here is that the UUID, as well as the public and private key, will be reset, and you will have to configure these values again in your frontend's production environment.

In the future, I recommend using a mosparo project for testing and a mosparo project for the production environment.

Kind regards,

zepich

**Reply by 0MAL** [reply]
I think the option 1 is enough for my case.
